### PR TITLE
Fix for https://github.com/TwistedInteractive/author_roles/issues/25

### PR DESCRIPTION
**Pull request by nathanhornby**
Provided by @nitriques

### CODE REVIEW

**Comment by nathanhornby** [comment]
Oh wait, hang on, looks like you've already done the fix, just with a slightly different approach - I wasn't paying attention when I pasted over the function!

If I'm mistaken just let me know.

**Comment by nitriques** [comment]
Yup you're missing the last if statement :smile:

**Comment by nathanhornby** [comment]
Wait, me or @brendo?

**Comment by nitriques** [comment]
Sorry I misread it, this PR seems ok!

**Comment by nitriques** [comment]
No it does not.

@brendo approach to use continue is better, but it still missing the first if statement.

File: extension.driver.php [hunk 1]
```diff
@@ -451,24 +451,26 @@ public function makePreAdjustements($context) {
...
 	// names can be a comma seperated list or an array of strings
-	private static function findChildren(XMLElement $element, $names) {
-		if(!is_array($names)) {
-			$names = explode(',', $names);
-		}
-
-		$children = array();
-
-		foreach($element->getChildren() as $child) {
-			if(!($child instanceof XMLElement)) continue;
```
Comment (nitriques) [inline extension.driver.php:462]: This should be kept

File: extension.driver.php [hunk 2]
```diff
@@ -451,24 +451,26 @@ public function makePreAdjustements($context) {
...
+	        if(!is_array($names)) {
+	        	$names = explode(',', $names);
+	        }
+	
+	        $children = array();
+	
+	        if (!($element instanceof XMLElement)) {
+	        	return $children;
+	        }
+
```
Comment (nitriques) [inline extension.driver.php:464]: This if statement is missing.

**Comment by nathanhornby** [comment]
@nitriques 

Isn't that what this is doing? (`461`)

```
foreach($element->getChildren() as $child) {
    if(!($child instanceof XMLElement)) continue;
```

**Comment by nathanhornby** [comment]
For what it's worth the `integration` branch of this fork does seem to be working fine without the patch.

**Comment by nitriques** [comment]
> For what it's worth the integration branch of this fork does seem to be working fine without the patch.

Great!

**Comment by nitriques** [comment]
`461` is differente from 

``` php
if (!($element instanceof XMLElement)) {
        return $children;
}
```

Both are required in order to bullet proof the function.

**Pull request closed without merging**